builtin/credential/aws-ec2/path_login.go

```diff
@@ -265,60 +319,96 @@ func (b *backend) parseIdentityDocument(s logical.Storage, pkcs7B64 string) (*id
 // option is enabled on the registered role.
 func (b *backend) pathLoginUpdate(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	identityDocB64 := data.Get("identity").(string)
+	var identityDocBytes []byte
+	var err error
+	if identityDocB64 != "" {
+		identityDocBytes, err = base64.StdEncoding.DecodeString(identityDocB64)
+		if err != nil || len(identityDocBytes) == 0 {
+			return logical.ErrorResponse("failed to base64 decode the instance identity document"), nil
+		}
+	}
 
-	pkcs7B64 := data.Get("pkcs7").(string)
-	if pkcs7B64 == "" {
-		return logical.ErrorResponse("missing pkcs7"), nil
+	signatureB64 := data.Get("signature").(string)
+	signature := ""
+	if signatureB64 != "" {
+		signatureBytes, err := base64.StdEncoding.DecodeString(signatureB64)
+		if err != nil {
+			return logical.ErrorResponse("failed to base64 decode the SHA256 RSA signature of the instance identity document"), nil
+		}
+		signature = string(signatureBytes)
```

Comment:
Oh, sorry for this. I worked on that suggestion and modified one parameter and left the other as it is. I'll change this.